### PR TITLE
Fix response to /rooms/{roomId}/join v2

### PR DESCRIPTION
**Pull request by anoadragon453**
Continuation of #684 but merged-forward.

Also did a little code cleanup and added a new, passing test to the testfile.

### CODE REVIEW

**Comment by Cnly** [comment]
Does this obsolete #722?

**Comment by anoadragon453** [comment]
Indeed it does, thanks :)

File: clientapi/routing/membership.go [hunk 1]
```diff
@@ -215,3 +209,41 @@ func getMembershipStateKey(
 
 	return
 }
+
+func checkAndProcessThreepid(
+	req *http.Request,
+	device *authtypes.Device,
+	body *threepid.MembershipRequest,
+	cfg config.Dendrite,
```
Comment (Cnly) [inline clientapi/routing/membership.go]: It seems a lot of places are passing `cfg` not as a pointer. Maybe they need to be changed at some point?

Comment (anoadragon453) [reply clientapi/routing/membership.go]: Mm, indeed. I don't believe there's any reason not too, and there's RAM savings in doing so.

Comment (anoadragon453) [reply clientapi/routing/membership.go]: https://github.com/matrix-org/dendrite/issues/736

**Pull request merged**